### ISSUE TITLE
Move the `DefaultTransform` logic to AeMCMC

### ISSUE THREAD

**Issue by rlouf** [aesara-devs/aeppl#201]
The transformations that should be applied are sampler-specific, so automation does not belong to AePPL.

**Comment by brandonwillard** [comment]
Is the basic idea that we should generate log-probabilities with AePPL and apply the transforms to those graphs using AeMCMC, instead of doing all that together in AePPL?

**Comment by rlouf** [comment]
Yes. AeMCMC would transform the graph and specify the variables to condition on. But that might only make sense when AePPL is able to condition on transformed variables e.g. `at.exp(x_rv)` directly.
